ecosystem/indexer/src/models/token.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{models::events::Event, schema::tokens};
+use aptos_rest_client::types;
+use std::{collections::HashMap, fmt, fmt::Formatter, str::FromStr};
+
+use serde::{Deserialize, Deserializer, Serialize};
+
+#[derive(Associations, Debug, Identifiable, Insertable, Queryable, Serialize)]
+#[diesel(table_name = "tokens")]
+#[primary_key(token_id)]
+pub struct Token {
+    pub token_id: String,
+    pub creator: String,
+    pub collection: String,
+    pub name: String,
+    pub description: String,
+    pub max_amount: Option<i64>,
+    pub supply: i64,
+    pub uri: String,
```

Comment:
This is the arweave / IPFS URI correct?

Reply:
Yes, this is a URI that we can use for either arweave  or IPFS